### DISCUSSION TITLE
Webcam and microphone bandwidth.

### DISCUSSION THREAD

**Discussion by Therealjosephchrzempiec** [networked-aframe/networked-aframe#482] [General]
Hello, I was wondering how much processing and bandwidth on the client size does it take to do audio and microphone? I'm okay on the server side but it's the client I'm wondering about. 

Joseph

**Comment by vincentfretin** [comment]
Just a single audio track playing is really not much, you probably can't see the difference of the cpu usage before the call and during the call. Transmitting the audio stream via WebRTC with opus codec is typically 20-40 Kbps

**Reply by Therealjosephchrzempiec** [reply]
But what about wecams if we wanted to do that?

**Reply by vincentfretin** [reply]
It depends on the video resolution that you specify when using getUserMedia and the WebRTC sdp negotiation with the server for the video codec (could be h264, vp8 or vp9). The h264 video codec have hardware acceleration for encoding and decoding on practically all devices, for vp8/vp9 some old devices don't have hardware acceleration for it, so encoding/decoding is run in software so take a bit much of the cpu.
A 720p video is between 1.5 to 3 Mbps, a 480p is between 0.5 to 1.5 Mbps.

**Reply by Therealjosephchrzempiec** [reply]
On mobile it should be that bad. Also Is there a way to set that limit to a 480p or 540p?

**Reply by vincentfretin** [reply]
On mobile if you're not careful, you can indeed have a really high resolution transferred over WebRTC. You need to add constraints when calling getUserMedia, see the [ShareCameraButton in naf-valid-avatars example](https://github.com/networked-aframe/naf-valid-avatars/blob/a50921f732f929f88f0fd6e4ec16940ac9be94cd/src/ShareScreenButton.tsx#L146-L157) that is used only with the janus adapter at the time of this writing.
For easyrtc you can't currently configure it, see #300
